src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -187,87 +182,81 @@ public void Dispose()
         }
 
         /// <summary>
-        /// This function will evaluate given string and it will try to convert
-        /// it to a Uri object. The string could contain URI fragment, IP address and  port
-        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// This function is used to parse WinINet Proxy strings. The strings are a semicolon
+        /// or whitespace separated list, with each entry in the following format:
+        /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
         /// </summary>
-        private static Uri GetUriFromString(string value)
+        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy)
         {
+            insecureProxy = null;
+            secureProxy = null;
             if (string.IsNullOrEmpty(value))
             {
-                return null;
+                return;
             }
 
-            if (!value.Contains("://"))
+            if (value.IndexOfAny(s_proxyDelimiters) == -1)
             {
-                value = "http://" + value;
+                ParseProxyConfigPart(value, ref insecureProxy, ref secureProxy);
+                return;
             }
 
-            if (Uri.TryCreate(value, UriKind.Absolute, out Uri uri) &&
-                    (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps))
+            string[] parts = value.Split(s_proxyDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            foreach (string part in parts)
             {
-                // We only support http and https for now.
-                return uri;
+                ParseProxyConfigPart(part, ref insecureProxy, ref secureProxy);
+
+                if (insecureProxy != null && secureProxy != null)
+                {
+                    return;
+                }
             }
-            return null;
         }
 
-        /// <summary>
-        /// This function is used to parse WinINet Proxy strings. The strings are a semicolon
-        /// or whitespace separated list, with each entry in the following format:
-        /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
-        /// </summary>
-        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy )
+        private static void ParseProxyConfigPart(ReadOnlySpan<char> part, ref Uri insecureProxy, ref Uri secureProxy)
         {
-            secureProxy = null;
-            insecureProxy = null;
-            if (string.IsNullOrEmpty(value))
+            ref Uri destUri = ref insecureProxy;
```

Comment:
Why does this need to use 'ref' on both the left and right side of the assignment operator?  Haven't seen much use of this pattern in our codebase.

Reply:
[ref local](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/ref-returns) per C# 7.0/7.3.